pkg/spoke/submarineragent/config_controller_test.go

```diff
@@ -52,7 +55,7 @@ func testWorkerNodeLabeling(t *configControllerTestDriver) {
 
 	When("the desired number of gateway nodes are already", func() {
 		BeforeEach(func() {
-			t.nodes[0].Labels["submariner.io/gateway"] = "true"
+			t.nodes[0].Labels["submariner.io/gateway"] = true_str
```

Comment:
I don’t see the value here. The idea behind `goconst` is to ease the maintenance burden; a key indicator in my mind is whether or not the string being replaced can be given a name which reflects its _purpose_ rather than its value.

I’m curious why `goconst` doesn’t complain about the strings used as keys here, since they’re repeated too...

If we want to fix the repetitions in this set of functions, I would prefer extracting functions to set the labels:

```suggestion
			labelGateway(t.nodes[0])
```

Reply:
I made the Gateway-labeling function, and extracted all of those usages.

Goconst still thinks there are too many `true` values, however:

```
pkg/spoke/submarineragent/config_controller_test.go:48:41: string `true` has 4 occurrences, make it a constant (goconst)
	node.Labels["submariner.io/gateway"] = "true"
```